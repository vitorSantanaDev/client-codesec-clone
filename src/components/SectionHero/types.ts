import { ISectionHeroProps } from 'interfaces/sections'

export type IViewProps = ISectionHeroProps
