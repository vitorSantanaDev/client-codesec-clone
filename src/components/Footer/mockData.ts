export const mockDataFooterProps = {
  logo: {
    altText: 'logo codesec',
    urlImage: '/images/logo-2-codesec.svg'
  },
  footerContent: 'Todos od direitos reservados',
  links: [
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/instagram.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/facebook.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/youtube.svg'
      }
    }
  ]
}
