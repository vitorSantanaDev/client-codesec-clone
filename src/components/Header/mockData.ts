import { IViewProps } from './type'

export const mockHeaderProps: IViewProps = {
  logo: {
    urlImage: '/images/codesec-logo.svg',
    altText: 'Logo codesec'
  },
  button: {
    url: '/',
    buttonText: 'Inscreva-se'
  }
}
