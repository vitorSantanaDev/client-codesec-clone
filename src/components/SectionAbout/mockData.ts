export const mockSectionAboutProps = {
  subtitle: 'infos',
  title: 'About web series',
  description:
    'Aprenda PHP do Jeito Certo Utilizando 100% de Tecnologia Moderna e os Melhores Padr√µes de Projeto Nessa Webserie 100% Online e GRATUITA!',
  topics: [
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    }
  ],
  illustration: {
    urlImage: '/images/about-ilustration.svg',
    altText: 'Illustration about'
  }
}
