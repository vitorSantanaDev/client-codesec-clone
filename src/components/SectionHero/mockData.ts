export const mockSectionHeroProps = {
  subtitle: 'Subtitle Example content',
  subtitleIcon: true,
  subtitleIconUrl: '/images/icon-key.svg',
  title: 'Aprenda Web Hacking do zero ao avan√ßado',
  description: 'Lorem ipsum dolor sit amet',
  linkIcon: '/images/icon-eye.svg',
  linkText: 'Ler mais',
  link: '/'
}
