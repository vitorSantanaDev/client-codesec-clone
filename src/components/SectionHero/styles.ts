import styled, { css } from 'styled-components'
import { Wrapper as CompositionElement } from 'components/CompositionElement/styles'
import { Wrapper as Heading } from 'components/Heading/styles'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    height: 90vh;
    display: grid;
    align-items: center;
    gap: ${theme.spacings.medium};
    grid-template-columns: 3fr 1fr 2fr;

    @media ${theme.media.ltexLarge} {
      justify-content: center;
      justify-items: center;
      grid-template-columns: 1fr;
    }

    ${CompositionElement} {
      transform: rotate(270deg);

      @media ${theme.media.ltexLarge} {
        transform: rotate(360deg);
      }

      @media ${theme.media.lteSmall} {
        width: 325px;
      }
    }
  `}
`

export const ContentWrapper = styled.div`
  ${({ theme }) => css`
    ${Heading} {
      font-size: calc(${theme.font.sizes.xxlarge} - 0.2rem);
      margin: ${theme.spacings.small} 0;

      @media ${theme.media.lteMedium} {
        font-size: calc(${theme.font.sizes.large} + 0.4rem);
      }
    }
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    line-height: 23.4px;
    color: ${theme.colors.gray7};
    margin-bottom: ${theme.spacings.small};
    font-size: ${theme.font.sizes.small};
  `}
`

export const LinkElement = styled.a``

export const FormWrapper = styled.div``
