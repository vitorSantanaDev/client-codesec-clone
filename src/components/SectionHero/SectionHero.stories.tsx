import { ComponentStory, ComponentMeta } from '@storybook/react'
import SectionBackground from 'components/SectionBackground'
import SectionHero from '.'
import { mockSectionHeroProps } from './mockData'

export default {
  title: 'SectionHero',
  component: SectionHero,
  args: {
    ...mockSectionHeroProps
  }
} as ComponentMeta<typeof SectionHero>

export const Default: ComponentStory<typeof SectionHero> = (args) => (
  <SectionBackground backgroundImagePath="/images/bg-hero.png">
    <SectionHero {...args} />
  </SectionBackground>
)
