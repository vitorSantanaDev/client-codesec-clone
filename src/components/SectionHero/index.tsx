import React from 'react'
import Link from 'next/link'

import SectionBackground from 'components/SectionBackground'
import CompositionElement from 'components/CompositionElement'
import Container from 'components/Container'
import Form from 'components/Form'
import Heading from 'components/Heading'
import Subtitle from 'components/Subtitle'
import * as S from './styles'

type SectionHeroProps = {
  _id?: string
  subtitle: string
  title: string
  linkIcon: string
  link: string
  linkText: string
  description: string
  subtitleIcon?: boolean
  subtitleIconUrl?: string
}

const SectionHero = ({
  subtitle,
  subtitleIcon,
  subtitleIconUrl,
  title,
  description,
  linkIcon,
  linkText,
  link
}: SectionHeroProps) => {
  return (
    <SectionBackground backgroundImagePath="/images/bg-hero.png">
      <Container>
        <S.Wrapper>
          <S.ContentWrapper data-aos="fade-up" data-aos-duration="1500">
            <Subtitle icon={subtitleIcon} iconUrl={subtitleIconUrl}>
              {subtitle}
            </Subtitle>
            <Heading>{title}</Heading>
            <S.Description>{description}</S.Description>
            <Link href={link || ''}>
              <S.LinkElement>
                <Subtitle icon={true} iconUrl={linkIcon}>
                  {linkText}
                </Subtitle>
              </S.LinkElement>
            </Link>
          </S.ContentWrapper>
          <S.CompositionElementWrapper
            data-aos="fade-up"
            data-aos-duration="2000"
          >
            <CompositionElement />
          </S.CompositionElementWrapper>
          <S.FormWrapper data-aos="fade-up" data-aos-duration="2500">
            <Form
              srcImage="/images/logo-2-codesec.svg"
              altText="Logo codesec"
              url="/"
              subtitle="o acesso serÃ¡ enviado em seu e-mail"
              handleSubmit={() => {
                console.log('')
              }}
            />
          </S.FormWrapper>
        </S.Wrapper>
      </Container>
    </SectionBackground>
  )
}

export default SectionHero
