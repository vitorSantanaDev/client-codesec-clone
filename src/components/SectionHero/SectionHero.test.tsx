import { screen } from '@testing-library/react'
import { renderTheme } from 'styles/theme/render-theme'
import SectionHero from '.'

import { mockSectionHeroProps } from './mockData'

describe('<SectionHero />', () => {
  it('should render section hero with correct elements', () => {
    renderTheme(<SectionHero {...mockSectionHeroProps} />)
    const heading = screen.getByRole('heading', {
      name: /aprenda Web Hacking do zero ao avanÃ§ado/i
    })
    const description = screen.getByText(/lorem ipsum dolor sit amet/i)
    const subtitle = screen.getByText(/subtitle Example content/i)

    expect(heading).toBeInTheDocument()
    expect(description).toBeInTheDocument()
    expect(subtitle).toBeInTheDocument()
  })
  it('should render section hero with form', () => {
    renderTheme(<SectionHero {...mockSectionHeroProps} />)
    const logo = screen.getByRole('img', { name: /logo codesec/i })
    const button = screen.getByRole('button', { name: /quero me cadastrar/i })
    const inputName = screen.getByPlaceholderText(/seu nome/i)
    const inputEmail = screen.getByPlaceholderText(/seu email/i)

    expect(logo).toBeInTheDocument()
    expect(button).toBeInTheDocument()
    expect(inputName).toBeInTheDocument()
    expect(inputEmail).toBeInTheDocument()
  })
  it('should render section hero with link', () => {
    renderTheme(<SectionHero {...mockSectionHeroProps} />)
    const link = screen.getByText(/ler mais/i)
    expect(link).toBeInTheDocument()
  })
  it('should to match snapshot', () => {
    const { container } = renderTheme(<SectionHero {...mockSectionHeroProps} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
