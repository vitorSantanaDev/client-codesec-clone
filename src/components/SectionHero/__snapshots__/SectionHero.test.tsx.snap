// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionHero /> should to match snapshot 1`] = `
.c11 {
  width: 352px;
  height: 4px;
  background: #edeff0;
  box-shadow: 0px 0px 177px 4px rgba(232,55,82,0.7), 0px 0px 96px 18px rgba(232,55,82,0.5), 0px 0px 14px 1px rgba(232,55,82,0.8),0px 0px 4px 1px #e83752;
  border-radius: 8px;
}

.c1 {
  width: 100%;
  margin: 0 auto;
  max-width: 1200px;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c18 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c24 {
  width: 100%;
  max-width: 330px;
  padding: calc(1.6rem - 0.4rem);
  font-weight: 600;
  text-transform: unset;
  font-size: 1.6rem;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  color: #FFFFFF;
  background: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  -webkit-transition: all ease-in-out 300ms;
  transition: all ease-in-out 300ms;
}

.c24:hover {
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  box-shadow: -10px 15px 40px -15px #FA5770;
}

.c7 {
  font-size: 5.6rem;
  color: white;
  font-weight: 600;
  font-family: 'Raleway',sans-serif;
}

.c16 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c22 {
  width: 100%;
  display: block;
  padding: calc(1.6rem - 0.4rem) 0 calc(1.6rem - 0.4rem) 4.0rem;
  border: none;
  outline: none;
  border-radius: 4px;
  font-size: 1.6rem;
  color: #A3A3A3;
  background-color: #1C1B1B;
}

.c22:focus {
  background-color: #121111;
}

.c19 > svg {
  position: absolute;
}

.c17 > svg {
  position: absolute;
  top: 19px;
  left: 10px;
}

.c20 {
  display: inline-block;
  margin-bottom: calc(1.6rem - 1rem);
  font-weight: 400;
  font-size: calc(1.6rem - 0.2rem);
  color: #EB3D3D;
}

.c13 {
  cursor: pointer;
}

.c5 {
  background-image: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 700;
  -webkit-letter-spacing: 15%;
  -moz-letter-spacing: 15%;
  -ms-letter-spacing: 15%;
  letter-spacing: 15%;
  text-transform: uppercase;
  font-size: calc(1.6rem - 0.2rem);
}

.c5::before {
  content: '';
  width: 24px;
  height: 24px;
  border-radius: 50%;
  display: inline-block;
  background-size: cover;
  background-image: url(/images/icon-key.svg);
  margin-right: calc(1.6rem - 1rem);
}

.c9 {
  background-image: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 700;
  -webkit-letter-spacing: 15%;
  -moz-letter-spacing: 15%;
  -ms-letter-spacing: 15%;
  letter-spacing: 15%;
  text-transform: uppercase;
  font-size: calc(1.6rem - 0.2rem);
}

.c9::before {
  content: '';
  width: 24px;
  height: 24px;
  border-radius: 50%;
  display: inline-block;
  background-size: cover;
  background-image: url(/images/icon-eye.svg);
  margin-right: calc(1.6rem - 1rem);
}

.c25 {
  background-image: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 700;
  -webkit-letter-spacing: 15%;
  -moz-letter-spacing: 15%;
  -ms-letter-spacing: 15%;
  letter-spacing: 15%;
  text-transform: uppercase;
  font-size: calc(1.6rem - 0.2rem);
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 .c4 {
  margin-top: calc(1.6rem - 0.2rem);
}

.c12 .c6 {
  margin-top: calc(1.6rem - 0.8rem);
}

.c14 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c14 .c23 {
  max-width: 350px;
  margin-top: calc(1.6rem - 0.4rem);
  margin-bottom: 1.6rem;
}

.c15 {
  margin-bottom: 1.6rem;
}

.c15 .c21 {
  width: 350px;
}

.c2 {
  padding: calc(8rem + 7rem) calc(2.4rem - 1.4rem);
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2.4rem;
  grid-template-columns: 3fr 1fr 2fr;
}

.c2 .c10 {
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
}

.c3 .c6 {
  font-size: calc(4.8rem - 0.2rem);
  margin: 1.6rem 0;
}

.c8 {
  line-height: 23.4px;
  color: #A3A3A3;
  margin-bottom: 1.6rem;
  font-size: 1.6rem;
}

.c0 {
  width: 100%;
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  background-image: url(/images/bg-hero.png);
}

@media (max-width:768px) {
  .c7 {
    font-size: 4.0rem;
  }
}

@media screen and (max-width:560px) {
  .c12 .c4 {
    font-size: calc(1.6rem - 0.4rem);
  }
}

@media screen and (max-width:560px) {
  .c12 .c6 {
    font-size: 2.4rem;
  }
}

@media (max-width:420px) {
  .c14 .c23 {
    width: 325px;
  }
}

@media (max-width:420px) {
  .c15 .c21 {
    width: 325px;
  }
}

@media (max-width:912px) {
  .c2 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    justify-items: center;
    grid-template-columns: 1fr;
  }
}

@media (max-width:912px) {
  .c2 .c10 {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@media (max-width:420px) {
  .c2 .c10 {
    width: 325px;
  }
}

@media (max-width:768px) {
  .c3 .c6 {
    font-size: calc(3.2rem + 0.4rem);
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <span
          class="c4 c5"
          color="redGradient"
        >
          Subtitle Example content
        </span>
        <h1
          class="c6 c7"
          color="white"
        >
          Aprenda Web Hacking do zero ao avan√ßado
        </h1>
        <p
          class="c8"
        >
          Lorem ipsum dolor sit amet
        </p>
        <a
          class=""
        >
          <span
            class="c4 c9"
            color="redGradient"
          >
            Ler mais
          </span>
        </a>
      </div>
      <div
        class="c10 c11"
      />
      <div
        class=""
      >
        <form
          class="c12"
        >
          <div
            class=""
          >
            <img
              alt="Logo codesec"
              class="c13"
              src="/images/logo-2-codesec.svg"
            />
          </div>
          <div
            class="c14"
          >
            <div
              class="c15"
            >
              <div
                class="c16"
              >
                <span
                  class="c17"
                >
                  <svg
                    aria-hidden="true"
                    class="c18"
                    color="#E83752"
                    fill="currentColor"
                    focusable="false"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      data-name="Layer 2"
                    >
                      <g
                        data-name="person"
                      >
                        <path
                          d="M12 11a4 4 0 10-4-4 4 4 0 004 4zm0-6a2 2 0 11-2 2 2 2 0 012-2zM12 13a7 7 0 00-7 7 1 1 0 002 0 5 5 0 0110 0 1 1 0 002 0 7 7 0 00-7-7z"
                        />
                      </g>
                    </g>
                  </svg>
                </span>
                <div
                  class=""
                >
                  <span
                    class="c19"
                  />
                  <span
                    class="c20"
                  />
                </div>
              </div>
              <input
                class="c21 c22"
                id="name"
                name="name"
                placeholder="Seu nome"
                type="text"
                value=""
              />
              <div
                class="c16"
              >
                <span
                  class="c17"
                >
                  <svg
                    aria-hidden="true"
                    class="c18"
                    color="#E83752"
                    fill="currentColor"
                    focusable="false"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      data-name="Layer 2"
                    >
                      <path
                        d="M13 2a10 10 0 00-5 19.1 10.15 10.15 0 004 .9 10 10 0 006.08-2.06 1 1 0 00.19-1.4 1 1 0 00-1.41-.19A8 8 0 1112.77 4 8.17 8.17 0 0120 12.22v.68a1.71 1.71 0 01-1.78 1.7 1.82 1.82 0 01-1.62-1.88V8.4a1 1 0 00-1-1 1 1 0 00-1 .87 5 5 0 00-3.44-1.36A5.09 5.09 0 1015.31 15a3.6 3.6 0 005.55.61A3.67 3.67 0 0022 12.9v-.68A10.2 10.2 0 0013 2zm-1.82 13.09A3.09 3.09 0 1114.27 12a3.1 3.1 0 01-3.09 3.09z"
                        data-name="at"
                      />
                    </g>
                  </svg>
                </span>
                <div
                  class=""
                >
                  <span
                    class="c19"
                  />
                  <span
                    class="c20"
                  />
                </div>
              </div>
              <input
                class="c21 c22"
                id="email"
                name="email"
                placeholder="Seu Email"
                type="email"
                value=""
              />
            </div>
            <button
              class="c23 c24"
            >
              Quero me cadastrar
            </button>
          </div>
          <svg
            aria-hidden="true"
            class="c18"
            color="#E83752"
            fill="currentColor"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
            >
              <g
                data-name="alert-circle"
              >
                <path
                  d="M12 2a10 10 0 1010 10A10 10 0 0012 2zm0 18a8 8 0 118-8 8 8 0 01-8 8z"
                />
                <circle
                  cx="12"
                  cy="16"
                  r="1"
                />
                <path
                  d="M12 7a1 1 0 00-1 1v5a1 1 0 002 0V8a1 1 0 00-1-1z"
                />
              </g>
            </g>
          </svg>
          <span
            class="c4 c25"
            color="redGradient"
          >
            o acesso ser√° enviado em seu e-mail
          </span>
        </form>
      </div>
    </div>
  </div>
</div>
`;
